File: Algoritmer/MatrixMath.java
package Algoritmer;
import java.util.ArrayList;

public class MatrixMath {
    public static ArrayList<ArrayList<Integer>> multiply(ArrayList<ArrayList<Integer>> A, 
                                                         ArrayList<ArrayList<Integer>> B) {
        int rowsA = A.size();
        int colsA = A.get(0).size();
        int colsB = B.get(0).size();

        ArrayList<ArrayList<Integer>> result = new ArrayList<>();
        
        for (int i = 0; i < rowsA; i++) {
            ArrayList<Integer> newRow = new ArrayList<>();
            for (int j = 0; j < colsB; j++) {
                int sum = 0;
                for (int k = 0; k < colsA; k++) {
                    // Hämtar elementen och multiplicerar
                    sum += A.get(i).get(k) * B.get(k).get(j);
                }
                newRow.add(sum);
            }
            result.add(newRow);
        }
        return result;
    }
}

File: Algoritmer/MatrixMath.py
def MatrixMath(A, B):
    # A: n x m, B: m x p => Resultat: n x p
    rows_A = len(A)
    cols_A = len(A[0])
    cols_B = len(B[0])
    
    
    result = [[0 for _ in range(cols_B)] for _ in range(rows_A)]
    
    for i in range(rows_A):          # Gå igenom rader i A
        for j in range(cols_B):      # Gå igenom kolumner i B
            for k in range(cols_A):  # Matcha element (rad i A mot kolumn i B)
                result[i][j] += A[i][k] * B[k][j]
                
    return result